### ISSUE TITLE
Add function to remove singleton from edges

### ISSUE THREAD

**Issue by maximelucas** [ComplexGroupInteractions/xgi#29]
Currently, edges contain singletons (nodes) : 

```python
list(H._edge.values())
# [[0], [3, 4], [4], [4]]
```
it's often useful to have a list of edges that does not contain nodes.
Add a function to get that. 
Should the function output a new external list, of remove singletons from the edges stored internally in the HyperGraph?

**Comment by leotrs** [comment]
My suggestion is to implement the following:
```python
H.edges(order=1)
# -> EdgeView([[0], [3], ...])
H.singleton_edges()
# -> same as above
H.remove_singletons()
# -> modifies H and returns None
```
Similarly, we should do something similar for removing all connected components other than the largest one:
```python
H.connected_components()
# -> list of EdgeView objects
H.connected_components(k)
# -> list of EdgeView objects containing only the k largest components
H.largest_component()
# -> same as H.connected_components(1)
H.remove_components(k=1)
# -> modifies H by removing all but the k (default k=1) largest components and returns None
```

**Comment by nwlandry** [comment]
Fixed with #35
